design/ui/pagination/README.mdx

```diff
@@ -0,0 +1,170 @@
+---
+name: Pagination
+---
+
+import semanticUiCssOffline from 'semantic-ui-css-offline';
+import { Playground, Props } from 'docz';
+import { Pagination } from './es';
+
+# Espressive UI Pagination
+This component abstracts the pagination logic into a reusable block that can be implemented anywhere pagination is required.
+
+## Separation of concerns
+The only concern of this component is to display a UI that allows the User to paginate through some content, the presentation of such content as well as its source are outside of the scope of this component.
+
+
+## Props
+| Name    | Default    | Type | Description |
+| --------|------------|------| ------------|
+|entityNamePlural|`Null`|String|Entitiy's name in plural form|
+|entityNameSingular|`Total`|String|Entitiy's name in singular form|
+|isLoading|`false`|Boolean|Specifies if the data is loading|
+|onPaginationChange|`noop`|Function|Upstream logic to be called on change|
+|limit|`10`|Number|Number of items to display per page|
+|page|`1`|Number|Current page number|
+|totalItems|`0`|Number|Total count of items to be paginated|
```

Comment:
I think leave this table for now... but essentially, this table should be generated automatically for us from static analysis of the component using the Props component.

The one thing I think we lose though... is that this spec forced us to make decisions about code and how the component would work before we started writing code. So, we need to figure out how we want to define spec moving forward

For this project since it is centered around UX as much as code, we should have a spec definition that also encompasses those ideas as well. That spec pattern only exists as an idea right now. I think it makes sense for that spec to also live in this repo somewhere. Possibly as a separate MD file that we can write things on, and one that does not change as we change the component code so we can A/B against it to see if we are still following what we wanted.

If we did have a spec MD file, we could actually locate it in the package folder. And it could be a separate tab on the doc page for a component. It could also potentially be used to track the progress of the components we are working on and what our needs are. @CaffeinatedAllie that could actually be what we use to generate a table of components and patterns, and their progress.